config/initializers/breakers.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require 'rx/configuration'
+
+# Read the redis config, create a connection and a namespace for breakers
+redis_config = Rails.application.config_for(:redis).freeze
+redis = Redis.new(redis_config['redis'])
+redis_namespace = Redis::Namespace.new('breakers', redis: redis)
+
+# The Rx::Configuration class contains the host and path for Rx requests.
+# Create a matcher proc that returns true for POST requests to that host and path.
+rx_path = URI.parse(Rx::Configuration.instance.base_path).path
+rx_host = URI.parse(Rx::Configuration.instance.base_path).host
+rx_matcher = proc do |request_env|
+  request_env.method == :post &&
+    request_env.url.host == rx_host &&
+    request_env.url.path =~ /^#{rx_path}/
+end
```

Comment:
@saneshark No reason to rush. The only real change I'm making to the Rx and SM Client classes is the addition of the middleware, so it shouldn't be any problem to merge.